aodncore/pipeline/files.py

```diff
@@ -478,31 +604,31 @@ def add(self, pipeline_file, deletion=False, overwrite=False):
         :param overwrite: :py:class:`bool` which, if True, will overwrite an existing matching file in the collection
         :return: :py:class:`bool` which indicates whether the file was successfully added
         """
-        validate_pipelinefile_or_string(pipeline_file)
+        self.member_validator(pipeline_file)
         validate_bool(deletion)
         validate_bool(overwrite)
 
-        if isinstance(pipeline_file, PipelineFile):
+        if isinstance(pipeline_file, self.member_class):
             fileobj = pipeline_file
         else:
             if not deletion and not os.path.isfile(pipeline_file):
                 raise MissingFileError("file '{src}' doesn't exist".format(src=pipeline_file))
-            fileobj = PipelineFile(pipeline_file, is_deletion=deletion)
+            fileobj = self.member_class(pipeline_file, is_deletion=deletion)
```

Comment:
I get an error here when trying to create a `RemotePipelineFileCollection`. 
```
  File "/sw/chef/src/python-aodndata/aodndata/moorings/products_handler.py", line 88, in preprocess
    input_file_collection = RemotePipelineFileCollection(input_files)
  File "/sw/chef/src/python-aodncore/aodncore/pipeline/files.py", line 950, in __init__
    super(RemotePipelineFileCollection, self).__init__(*args, **kwargs)
  File "/sw/chef/src/python-aodncore/aodncore/pipeline/files.py", line 578, in __init__
    self.add(f)
  File "/sw/chef/src/python-aodncore/aodncore/pipeline/files.py", line 616, in add
    fileobj = self.member_class(pipeline_file, is_deletion=deletion)
TypeError: __init__() got an unexpected keyword argument 'is_deletion'
```
Interestingly, `is_deletion` is technically _only_ relevant to a remote file, not a local one, but that's not the way the handler deals with deletions...